themes/src/main/resources/theme/keycloak-preview/account/resources/app/content/signingin-page/SigningInPage.tsx

```diff
@@ -0,0 +1,183 @@
+                                <DataListAction aria-labelledby='foo' aria-label='foo action' id='setPasswordAction'>
+                                    <Button variant='primary'onClick={()=> this.updatePassword.execute()}><Msg msgKey='update'/></Button>
+                                </DataListAction>
+                            </DataListItemRow>
+                        </DataList>
+                    </StackItem>
+                    <StackItem isFilled>
+                        <Title headingLevel={TitleLevel.h2} size='2xl'>
+                        <strong><Msg msgKey='twoFactorAuth'/></strong>
+                        </Title>
+                        <DataList aria-label='foo'>
+                            <DataListItemRow>
+                                <DataListAction aria-labelledby='foo' aria-label='foo action' id='twoFactorOnOff'>
+                                        <Switch
+                                            aria-label='twoFactorSwitch'
+                                            label={this.state.twoFactorEnabledText}
+                                            isChecked={this.state.twoFactorEnabled}
+                                            onClick={this.handleTwoFactorSwitch}  
+                                        />
+                                    </DataListAction>
+                                </DataListItemRow>
+                            <DataListItemRow>
+                                <DataListItemCells
+                                    dataListCells={[
+                                        <DataListCell key='mobileAuth'><Msg msgKey='mobileAuthDefault'/></DataListCell>
+                                    ]}/>
+                                {!this.state.isTotpConfigured && 
+                                    <DataListAction aria-labelledby='foo' aria-label='foo action' id='setMobileAuthAction'>
+                                        <Button isDisabled={!this.state.twoFactorEnabled} variant='primary' onClick={()=> this.setUpTOTP.execute()}><Msg msgKey='setUp'/></Button>
+                                    </DataListAction>
+                                }
+                                {this.state.isTotpConfigured &&
+                                    <DataListAction aria-labelledby='foo' aria-label='foo action' id='setMobileAuthAction'>
+                                        <ContinueCancelModal buttonTitle='remove'
+                                                            isDisabled={!this.state.twoFactorEnabled}
+                                                            modalTitle={Msg.localize('removeMobileAuth')}
+                                                            modalMessage={Msg.localize('stopMobileAuth')}
+                                                            onContinue={this.handleRemoveTOTP}
+                                            />
+                                    </DataListAction>
+                                }
```

Comment:
It's not possible to remove authenticator. Seems like `isTotpConfigured` is always set to false.

Reply:
Correct.  When I first coded this it worked.  Changes on the server side broke it.  Bruno is aware.